processing/src/main/java/io/druid/segment/QueryableIndexIndexableAdapter.java

```diff
@@ -385,6 +385,9 @@ public IndexedInts seek(String value)
           return new EmptyIndexedInts();
         }
         if (currVal == null) {
+          if(currIndex == dimSet.size()) {
+            return new EmptyIndexedInts();
+          }
```

Comment:
It handles the case where the only entry in dimSet is the null string; the seek needs to stop and return if the dimSet has been exhausted. The index == size check is there to distinguish between the initial state of the seeker and when it has retrieved the null string's entry (in both cases currVal is null)